### DISCUSSION TITLE
Security migration to IsGranted

### DISCUSSION THREAD

**Discussion by one1two2** [symfony/symfony#48878] [Q&A] [answered]
Hello in 6.2 there are new annotation to replace Security. It seems that IsGranted has power to replace Security using an expressions but I'm unable to do it. It seems I did something wrong.

I got controller something like this

```
    #[Route('/articles/{article}', name: 'patch_article', methods: ['patch'])]
    #[Security("article.getAuthor() === user")]
    public function patchAction(Article $article): JsonResponse
```

I would like to change it to something like this

    #[Route('/articles/{article}', name: 'patch_article', methods: ['patch'])]
    #[IsGranted(new Expression("article.getAuthor() === user"))]
    public function patchAction(Article $article): JsonResponse

But It seems isGranted does not know how to get article object from route. Do I have to use map entity or something? Thanks for your help.

**Comment by stof** [comment, marked as answer]
The ExpressionVoter does not give you access to controller arguments (because it does not have access to them). However, there is one variable that *is* available: the subject on which you vote (as the `subject` variable).
So the solution is to use the article as the subject being voted one instead of voting on null (which is the default subject):

```php
#[Route('/articles/{article}', name: 'patch_article', methods: ['patch'])]
#[IsGranted(new Expression("subject.getAuthor() === user"), subject: 'article')]
public function patchAction(Article $article): JsonResponse
```

**Reply by WebMamba** [reply]
Maybe we should update the docs. A lot of people were using these features, so a lot of people gonna search for how to do this, but there is nothing about it in the documentation.

**Comment by webda2l** [comment]
See https://github.com/symfony/symfony-docs/pull/17513

**Reply by WebMamba** [reply]
Thanks for the PR @webda2l ! 😍

**Reply by webda2l** [reply]
> Thanks for the PR @webda2l ! heart_eyes

No no it's not mine, it's the @HypeMC one (thanks to him!) ;)

But it helps this discussion until its merge, so I linked it.
